provider/user_data_test.go

```diff
@@ -0,0 +1,32 @@
+package provider
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+func TestUserDataSource_basic(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testFireHydrantIsSetup(t) },
+		ProviderFactories: defaultProviderFactories(),
+		Steps: []resource.TestStep{
+			{
+				Config: testUserDataSourceConfig_basic(),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttrSet("data.firehydrant_user.test_user", "id"),
+					resource.TestCheckResourceAttr(
+						"data.firehydrant_user.test_user", "email", "test-user@firehydrant.io"),
```

Comment:
I would appreciate some advice on how best to test these new data sources. The existing `data` modules all seem to be tested in conjunction with a `resource` module.  To test this, I would need to introduce mocks, which don't seem to exist in this repository.